packages/ui/Toggleable/useToggle.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+
+import useControlledState from '../useControlledState';
+
+import Toggle from './Toggle';
+
+/**
+ * Creates one instance of the class, `Ctor` with the provided `args`, for the life of the
+ * component.
+ *
+ * @param {Function} Ctor Class constructor
+ * @param  {...any} args Arguments to pass to the constructor
+ * @private
+ */
+function useClass (Ctor, ...args) {
+	const ref = React.useRef(null);
+	ref.current = ref.current || new Ctor(...args);
```

Comment:
Could we combine 16 and 17 lines into `const ref = React.useRef(new Ctor(...args));`?

Reply:
The reason for the separation is to avoid creating new classes. Functionally, I believe the result will be the same due to how `useRef` behaves but in your code sample above, a new instance of `Ctor` would be created on each render and then discarded.